backend/src/main/java/com/bakdata/conquery/sql/conversion/select/DateDistanceConverter.java

```diff
@@ -21,27 +22,41 @@ public DateDistanceConverter(DateNowSupplier dateNowSupplier) {
 	}
 
 	@Override
-	public Field<Integer> convert(DateDistanceSelect select, ConversionContext context) {
+	public SqlSelects convert(DateDistanceSelect select, SelectContext context) {
 
-		ChronoUnit timeUnit = select.getTimeUnit();
-		Name startDateColumnName = DSL.name(select.getColumn().getName());
-		Date endDate = getEndDate(context);
+		ConquerySelect dateDistanceSelect = new com.bakdata.conquery.sql.conversion.cqelement.concept.model.select.DateDistanceSelect(
+				dateNowSupplier,
+				select.getTimeUnit(),
+				context.getTables().rootTable(),
+				select.getColumn(),
+				context.getParentContext().getDateRestrictionRange(),
+				select.getLabel(),
+				context.getParentContext().getSqlDialect().getFunction()
+		);
 
-		return context.getSqlDialect().getFunction().dateDistance(timeUnit, startDateColumnName, endDate)
-					  .as(select.getLabel());
-	}
+		// TODO: @awildturtok How should we handle FIRST and LAST if there is no validity date?
```

Comment:
Wir sortieren bei hier aktuell nach validity date und wenn nicht vorhanden, dann nach PID. Passt das?

Reply:
Wir nehmen dann immer first/last nach encounter, da kannst du nichts machen

Reply:
Was genau meinst du mit encounter?

Reply:
einfach der erste der Wert einfliegt, ohne jeglichen definierten Bezug

Reply:
Wir brauchen leider zwingend bei den Datenbanken ein ORDER BY. Also würden wir dann nach primary column sortieren?

Reply:
Die primrary column ist in dem fall aber immer gleich, also macht es keinen Unterschied. Was ich sagen möchte ist, es ist undefined behaviour, was ihr macht ist egal.

Reply:
Alles klar 👍